functions/go/ensure-name-segment/README.md

```diff
@@ -0,0 +1,92 @@
+# ensure-name-segment
+
+### Overview
+
+<!--mdtogo:Short-->
+
+Ensure a name segment is part of the name.
+
+<!--mdtogo-->
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+If the desired name segment is already part of the name, it takes no actions.
```

Comment:
What will happen if user want to add suffix `-deploy` to name `resource-deployment`? I suppose the suffix will not be added?

Reply:
It won't be added in this case. This is going to be the limitation.

Reply:
Probably we should list these limitations in the readme

Reply:
Yes, I added it in https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/226/files#diff-3c677f0bba9846eff1255ea8e3d7773948e2d4832c4b4439e7e89745ad365c81R18